### PR TITLE
Fixed login button disabled issue. #2045

### LINKED ISSUE #2045: Active Login button only when 6 characters are entered in password field
**Actual Behaviour**

Login button becomes active even if one character is entered.

**Expected Behaviour**

Whenever user signs up, minimum length of password required is 6 characters. So in login form, "Login" button should only be active when 6 characters are entered.

**Steps to reproduce it**

1) Go to login form and enter 1 character in password field.

**Would you like to work on the issue?**

Yes, I would like to work on this.

### PR DESCRIPTION
**Pull request by plxity**
Fixes #2045 

Changes: Login button becomes active only when 6 characters are added.

Demo Link: https://pr-2046-fossasia-susi-web-chat.surge.sh

### CODE REVIEW

File: src/components/Auth/Login/Login.react.js [hunk 1]
```diff
@@ -317,7 +317,7 @@ class Login extends Component {
                     : '#19314B'
                 }
                 labelColor="#fff"
-                disabled={!isValid || loading}
+                disabled={!isValid || loading || this.state.password.length < 6}
```
Comment (AakashMallik) [inline src/components/Auth/Login/Login.react.js]: Don't put this here. Take reference from the code of Signup and see how we have handled this case.

Comment (plxity) [reply src/components/Auth/Login/Login.react.js]: @AakashMallik  Okay Sure.

**Comment by plxity** [comment]
@akshatnitd Sir, Please review this.

File: src/components/Auth/Login/Login.react.js [hunk 2]
```diff
@@ -192,9 +193,11 @@ class Login extends Component {
       }
       case 'password': {
         const password = event.target.value.trim();
+        const passwordError = !(password.length >= 6);
         this.setState({
           password,
           passwordErrorMessage: !password ? 'Enter a valid password' : '',
+          passwordLength: passwordError,
```
Comment (AakashMallik) [inline src/components/Auth/Login/Login.react.js]: I think an extra passwordLength var was not needed. What you can do is, give the user an error saying that the password length should be at least 6. So we can make use of just the ```passwordErrorMessage``` var to handle that case. I hope you get my point.

Comment (plxity) [reply src/components/Auth/Login/Login.react.js]: Yes, I got your point. 
But why to display the error if the password is lesser than 6 characters.
The button should automatically become active as 6 characters are entered. account.susi.ai works in the same manner. It doesn't display the error.

Comment (AakashMallik) [reply src/components/Auth/Login/Login.react.js]: Actually, we displayed an error in Signup so I thought the user should have a clear idea about why the button is not being activated. Up to you...

**Comment by plxity** [comment]
@fragm3 I don't think displaying error is a  right option. I'm still waiting for @akshatnitd sir review. If he suggests, I will make the changes.

**Comment by plxity** [comment]
@AakashMallik Yes sure. Actually, I had an exam that's why I couldn't work on this.

**Comment by plxity** [comment]
@akshatnitd Okay sure sir.

**Comment by akshatnitd** [comment]
@plxity Please finalise the PR

**Comment by plxity** [comment]
@akshatnitd Apologies sir for the delay. I will make the changes. :)

**Comment by plxity** [comment]
@akshatnitd  Please review and merge this sir. :)

**Comment by plxity** [comment]
@akshatnitd Please review this sir. :)

**Comment by plxity** [comment]
@akshatnitd Please review this sir. :)

File: src/components/Auth/Login/Login.react.js [hunk 3]
```diff
@@ -192,9 +195,14 @@ class Login extends Component {
       }
       case 'password': {
         const password = event.target.value.trim();
+        const passwordError = !(password.length >= 6);
```
Comment (fragm3) [inline src/components/Auth/Login/Login.react.js]: Make `passwordError`, `passwordLength`

File: src/components/Auth/Login/Login.react.js [hunk 4]
```diff
@@ -239,7 +247,9 @@ class Login extends Component {
       password,
       emailErrorMessage,
       passwordErrorMessage,
+      passwordLengthError,
```
Comment (fragm3) [inline src/components/Auth/Login/Login.react.js]: Rename `passwordLengthError` to passwordLengthErrorMessage

File: src/components/Auth/Login/Login.react.js (hunk 3 above)
Comment (fragm3) [reply src/components/Auth/Login/Login.react.js]: Instead use `const passwordLength = password.length < 6`

**Comment by plxity** [comment]
@fragm3 I have done the changes. Please review it once.

**Comment by plxity** [comment]
@anshumanv @akshatnitd  please review this. :)

File: src/components/Auth/Login/Login.react.js [hunk 5]
```diff
@@ -192,9 +194,13 @@ class Login extends Component {
       }
       case 'password': {
         const password = event.target.value.trim();
+        const passwordLength = !(password.length >= 6);
```
Comment (akshatnitd) [inline src/components/Auth/Login/Login.react.js]: `!(password.length >= 6)` is not equal to `passwordLength`. Please name it suitably or change the conditions.

Comment (plxity) [reply src/components/Auth/Login/Login.react.js]: @akshatnitd I have done the changes, sir. Please review it again. :)

File: src/components/Auth/Login/Login.react.js [hunk 6]
```diff
@@ -192,9 +194,13 @@ class Login extends Component {
       }
       case 'password': {
         const password = event.target.value.trim();
+        const passwordLength = password.length < 6;
```
Comment (akshatnitd) [inline src/components/Auth/Login/Login.react.js]: Again, the same issue. Does passwordLength mean true/false if the length of password is less than 6 or just the length of the password ? Naming of variable shouldn't be ambiguous.

**Comment by plxity** [comment]
@akshatnitd Sir, I have made the changes. I hope these changes are fine. Please review it again. :) Sorry for delaying it.

**Comment by harsh-jindal** [comment]
@plxity Squash your commits.

**Comment by plxity** [comment]
@akshatnitd Please review this. :)

**Comment by plxity** [comment]
@akshatnitd Sir, I have fixed this. Can you please review it once? :)

**Pull request merged**